Sources/Utility/Process.swift

```diff
@@ -112,148 +125,157 @@ public final class Process: ObjectIdentifierProtocol {
 
     /// Launch the subprocess.
     public func launch() throws {
-        assert(!launched, "It is not allowed to launch the same process object again.")
-        launched = true
-
-        // Initialize the spawn attributes.
-      #if os(macOS)
-        var attributes: posix_spawnattr_t? = nil
-      #else
-        var attributes = posix_spawnattr_t()
-      #endif
-        posix_spawnattr_init(&attributes)
-
-        // Unmask all signals.
-        var noSignals = sigset_t()
-        sigemptyset(&noSignals)
-        posix_spawnattr_setsigmask(&attributes, &noSignals)
-
-        // Reset all signals to default behavior.
-      #if os(macOS)
-        var mostSignals = sigset_t()
-        sigfillset(&mostSignals)
-        sigdelset(&mostSignals, SIGKILL)
-        sigdelset(&mostSignals, SIGSTOP)
-        posix_spawnattr_setsigdefault(&attributes, &mostSignals)
-      #else
-        // On Linux, this can only be used to reset signals that are legal to
-        // modify, so we have to take care about the set we use.
-        var mostSignals = sigset_t()
-        sigemptyset(&mostSignals);
-        for i in 1 ..< SIGUNUSED {
-            if i == SIGKILL || i == SIGSTOP {
-                continue
+        try self.serialQueue.sync {
+            assert(!_launched, "It is not allowed to launch the same process object again.")
+            _launched = true
```

Comment:
Only _launched should be in queue here since calling this method twice is a programming error, we should ask turn that assert into precondition.